webapp/channels/src/components/admin_console/settings_group.tsx

```diff
@@ -4,26 +4,33 @@
 import PropTypes from 'prop-types';
 import React from 'react';
 
-export default class SettingsGroup extends React.PureComponent {
-    static get propTypes() {
-        return {
-            show: PropTypes.bool.isRequired,
-            header: PropTypes.node,
-            title: PropTypes.node,
-            subtitle: PropTypes.node,
-            children: PropTypes.node,
-            container: PropTypes.bool,
-        };
-    }
+type Props = {
+    id?: string;
+    show: boolean;
+    header?: React.ReactNode;
+    title?: React.ReactNode;
+    subtitle?: React.ReactNode;
+    children?: React.ReactNode;
+    container?: boolean;
+};
 
-    static get defaultProps() {
-        return {
-            show: true,
-            container: true,
-        };
-    }
+export default class SettingsGroup extends React.PureComponent<Props> {
+    static propTypes = {
+        id: PropTypes.string,
+        show: PropTypes.bool.isRequired,
+        header: PropTypes.node,
+        title: PropTypes.node,
+        subtitle: PropTypes.node,
+        children: PropTypes.node,
+        container: PropTypes.bool,
+    };
 
-    render() {
+    static defaultProps = {
+        show: true,
+        container: true,
+    };
```

Comment:
This shouldn't be needed either:
https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/default_props/#you-may-not-need-defaultprops

Reply:
Thanks @larkox for the reference article. I updated the code and passed the required prop `show` from the two components that were initially relying on the `defaultProps` from `SettingsGroup`. Let me know if this is the correct approach.